File: app/src/main/java/com/example/sunddenfix/retrofit/business/HtcHostnameVerifier.java
package com.example.sunddenfix.retrofit.business;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.SSLSession;

/**
 * https域名验证器
 */
public class HtcHostnameVerifier implements HostnameVerifier {

    private String[] mHostUrls;

    public HtcHostnameVerifier(String[] hostUrls) {
        mHostUrls = hostUrls;
    }

    @Override
    public boolean verify(String hostname, SSLSession session) {
        boolean ret = false;
        for (String host : mHostUrls) {
            if (host.equalsIgnoreCase(hostname)) {
                ret = true;
            }
        }
        return ret;
    }
}

File: app/src/main/java/com/example/sunddenfix/retrofit/business/HtcHostnameVerifier.kt
package com.example.sunddenfix.retrofit.business

import javax.net.ssl.HostnameVerifier
import javax.net.ssl.SSLSession

/**
 * https域名验证器
 */
class HtcHostnameVerifier(private val mHostUrls: Array<String>) : HostnameVerifier {

    override fun verify(hostname: String, session: SSLSession): Boolean {
        var ret = false
        for (host in mHostUrls) {
            if (host.equals(hostname, ignoreCase = true)) {
                ret = true
            }
        }
        return ret
    }
}